afkak/common.py

```diff
@@ -85,12 +93,55 @@
 
 # Other useful structs
 OffsetAndMessage = namedtuple("OffsetAndMessage", ["offset", "message"])
-Message = namedtuple("Message", ["magic", "attributes", "key", "value"])
+
+
 TopicAndPartition = namedtuple("TopicAndPartition", ["topic", "partition"])
 SourcedMessage = namedtuple(
     "SourcedMessage", TopicAndPartition._fields + OffsetAndMessage._fields)
 
 
+class Message(namedtuple("Message", ["magic", "attributes", "key", "value"])):
+    """
+    A Kafka `message`_ in format 0.
+
+    :ivar int magic: Message format version, always 0.
+    :ivar int attributes: Compression flags.
+    :ivar bytes key:
+        Message key, or ``None`` when the message lacks a key.
+        Note that the key is required on a compacted topic.
+    :ivar bytes value:
+        Message value, or ``None`` if this is a tombstone a.k.a. null message.
+
+    .. message:: https://kafka.apache.org/documentation/#messageset
+    """
+    __slots__ = ()
+
+    def __repr__(self):
+        bits = ['<Message v0']
+
+        if self.attributes != 0:
+            if self.attributes == CODEC_GZIP:
+                codec = ' CODEC_GZIP'
+            elif self.attributes == CODEC_SNAPPY:
+                codec = ' CODEC_SNAPPY'
+            elif self.attributes == CODEC_LZ4:
+                codec = ' CODEC_LZ4'
+            else:
+                codec = ' attributes=0x{:x}'.format(self.attributes)
+            bits.append(codec)
+
+        if self.key is not None:
+            bits.append(' key={!r}'.format(self.key))
+
+        if self.value is None or len(self.value) < 1024:
```

Comment:
The choice to show up to 1023 bytes, and then if the message is 1024 only show 512 seems "odd" :-)
Also, showing the same message formatting (always show the length) for both might make programatic processing easier.